tests/legacy-cli/e2e/tests/mcp/registers-tools.ts

```diff
@@ -0,0 +1,41 @@
+import { chdir } from 'process';
+import { exec, ProcessOutput, silentNpm } from '../../utils/process';
+import assert from 'node:assert/strict';
+
+const MCP_INSPECTOR_PACKAGE_NAME = '@modelcontextprotocol/inspector-cli';
+const MCP_INSPECTOR_COMMAND_NAME = 'mcp-inspector-cli';
+
+async function runInspector(...args: string[]): Promise<ProcessOutput> {
+  const result = await exec(
+    MCP_INSPECTOR_COMMAND_NAME,
+    '--cli',
+    'npx',
+    '--no',
+    '@angular/cli',
+    'mcp',
+    ...args,
```

Comment:
Question: How is `npx` pulling in the locally built version of the CLI? Is this going through Verdaccio through an `.npmrc` configuration or something like that?

I'm wondering if there's a more direct implementation through something like: `mcp-inspector-cli --cli node_modules/.bin/ng mcp ...args`?

Reply:
`npx` uses the locally installed version if present, then the global.  If neither are present, it then tries to download from the registry.  However, the `--no` option prevents that last step.  The package should always be present so the failure with the `--no` option effectively acts as an assertion for the test setup.
The arguments are based on the recommended usage/configuration as well to better represent real world use.